data/questions_gen_postgres.csv

```diff
@@ -44,7 +44,7 @@ How many courses are offered for each semester id?,"SELECT course_offering.semes
 "What is the total number of students enrolled in each course, ordered from highest to lowest?","SELECT {course.course_id, course.name, course.number}, SUM(course.num_enrolled) AS total_students FROM course GROUP BY {} ORDER BY total_students DESC NULLS LAST;",advising,order_by,
 "What is the total number of credits earned by each student, ordered from highest to lowest? Give the student id and the total number of credits.","SELECT student.student_id, student.total_credit FROM student ORDER BY student.total_credit DESC NULLS LAST;",advising,order_by,
 "What is the name of the instructor who has taught the most courses, and how many courses have they taught?","SELECT instructor.name, count(offering_instructor.offering_id) AS num_courses FROM offering_instructor JOIN instructor ON offering_instructor.instructor_id = instructor.instructor_id GROUP BY instructor.name ORDER BY num_courses DESC LIMIT 1;",advising,order_by,
-What is the ratio of the total number of students enrolled in courses with exams to the total number of students enrolled in courses without exams?,"SELECT CAST(SUM(CASE WHEN course.has_exams THEN course.num_enrolled ELSE 0 END) AS FLOAT) / NULLIF(SUM(CASE WHEN NOT course.has_exams THEN course.num_enrolled ELSE 0 END), 0) AS ratio FROM course;",advising,ratio,
+What is the ratio of the total number of students enrolled in courses with exams to the total number of students enrolled in courses without exams?,"WITH exam_courses AS (SELECT gsi.student_id FROM gsi JOIN course_offering co ON gsi.course_offering_id = co.offering_id JOIN course c ON co.course_id = c.course_id WHERE c.has_exams = TRUE), no_exam_courses AS (SELECT gsi.student_id FROM gsi JOIN course_offering co ON gsi.course_offering_id = co.offering_id JOIN course c ON co.course_id = c.course_id WHERE c.has_exams = FALSE) SELECT CAST(COUNT(DISTINCT ec.student_id) AS FLOAT) / NULLIF(COUNT(DISTINCT nec.student_id), 0) AS ratio FROM exam_courses ec, no_exam_courses nec;",advising,ratio,

```

Comment:
this now correctly counts the number of students instead of the number of courses